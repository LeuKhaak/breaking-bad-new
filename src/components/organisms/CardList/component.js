import React from "react";
import * as styles from "./styles.module.scss";
import CardMini from "../../../components/molecules/CardMini";
import Loader from "../../../components/atoms/Loader";
import Error from "../../../components/atoms/Error";
import { Link } from "react-router-dom";

function CardList({ getError, loader, cardsData, tiles }) {
  return getError ? (
    <Error notice="The request failed!" />
  ) : loader ? (
    <Loader loader />
  ) : (
    <div className={styles.cardList}>
      <div className={styles.container}>
        <div className={tiles ? styles.noHeadersLine : styles.headersLine}>
          <span className={styles.status}>Status</span>
          <span className={styles.name}>Name</span>
          <span className={styles.birthday}>Birthday</span>
        </div>
        <div className={tiles ? styles.list : styles.listLines}>
          {cardsData &&
            cardsData.map((element) => (
              <Link
                key={element.char_id}
                to={`/person/${element.char_id}`}
                className={styles.link}
              >
                <CardMini
                  tiles={tiles}
                  name={element.name}
                  birthday={element.birthday}
                  img={element.img}
                  status={element.status}
                />
              </Link>
            ))}
        </div>
      </div>
    </div>
  );
} //

export default CardList;
