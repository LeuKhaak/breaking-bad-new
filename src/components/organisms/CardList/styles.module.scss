@import "../../../assets/styles/styles.scss";

.cardList {
  //min-height: calc(100vh - 10rem);
  .container {
    @include content;
    display: flex;
    flex-direction: column;
    align-items: center;
    @include tabletSmall {
      width: 95%;
    }
    .noHeadersLine {
      display: none;
    }
    .headersLine {
      padding-bottom: 1.5rem;
      width: 74.5rem;
      border-bottom: 0.0625rem solid $greyFontColor;
      font-size: 0.875rem;
      @include desktop {
        width: 72.5rem;
      }
      @include desktopSmall {
        width: 64rem;
      }
      @include tabletLandscape {
        width: 49rem;
      }
      @include tabletSmall {
        width: 48rem;
      }
      @include tabletSmallSecond {
        width: 36.5rem;
      }
      @include mobile {
        width: 27.5rem;
      }
      @include mobileSecond {
        width: 23rem;
        display: flex;
        justify-content: space-between;
      }
      @include mobileSmall {
        width: 18rem;
      }
      .status {
        margin: 0 10.25rem 0 6.5rem;
        @include tabletSmallSecond {
          margin: 0 4.5rem 0 5.5rem;
        }
        @include mobile {
          margin: 0 3.5rem 0 5.5rem;
        }
        @include mobileSecond {
          margin: 0;
        }
      }
      .name {
        margin-right: 15.95rem;
        @include tabletSmallSecond {
          margin-right: 10.75rem;
        }
        @include mobile {
          margin-right: 8.25rem;
        }
        @include mobileSecond {
          margin-right: 0;
        }
      }
    }
    .listLines {
      .link {
        text-decoration: none;
      }
    }
    .list {
      display: grid;
      grid-template-columns: 1fr 1fr 1fr;
      grid-gap: 1.5rem 2rem;
      @include desktop {
        grid-gap: 1.5rem 1rem;
      }
      @include desktopSmall {
        grid-template-columns: 1fr 1fr;
        grid-gap: 1.5rem 2rem;
      }
      @include tabletSmall {
        grid-gap: 1.5rem 1rem;
      }
      @include tabletSmallSecond {
        grid-template-columns: 1fr;
        grid-gap: 1.5rem;
      }
      .link {
        text-decoration: none;
      }
    }
  }
}
