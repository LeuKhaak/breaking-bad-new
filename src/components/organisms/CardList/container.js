import { useSelector, useDispatch } from "react-redux";
import { React, useEffect } from "react";
import CardList from "./component";
import { getStartData } from "../../../store/actions/person";

function CardListContainer() {
  const dispatch = useDispatch();

  const cardsData = useSelector((state) => state.cards.cardList);
  const getError = useSelector((state) => state.cards.getError);
  const loader = useSelector((state) => state.cards.loader);

  useEffect(() => {
    dispatch(getStartData());
  }, [dispatch]);

  return <CardList cardsData={cardsData} getError={getError} loader={loader} />;
}

export const container = CardListContainer;
