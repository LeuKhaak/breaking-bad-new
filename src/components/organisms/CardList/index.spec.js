import React from "react";
import { shallow } from "enzyme";
import CardList from "./component";

let component = null;
const setComponent = (props) => shallow(<CardList {...props} />);
beforeEach(() => {
  component = setComponent();
});

describe("CardList testing:", () => {
  test("is Error shown", () => {
    component = setComponent({
      getError: true,
    });
    expect(component).toMatchSnapshot();
  });

  test("is Loader shown", () => {
    component = setComponent({
      getError: null,
      loader: true,
    });
    expect(component).toMatchSnapshot();
  });

  test("is CardList rendered", () => {
    component = setComponent({
      getError: null,
      loader: false,
      tiles: true,
      cardsData: [
        {
          char_id: 1,
          name: "Bill",
        },
        {
          char_id: 2,
          name: "Cat",
        },
      ],
    });
    expect(component).toMatchSnapshot();
  });
});
