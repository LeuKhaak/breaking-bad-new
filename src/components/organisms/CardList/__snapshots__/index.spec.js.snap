// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardList testing: is CardList rendered 1`] = `
<div>
  <div>
    <div>
      <span>
        Status
      </span>
      <span>
        Name
      </span>
      <span>
        Birthday
      </span>
    </div>
    <div>
      <Link
        key="1"
        to="/person/1"
      >
        <CardMini
          name="Bill"
          tiles={true}
        />
      </Link>
      <Link
        key="2"
        to="/person/2"
      >
        <CardMini
          name="Cat"
          tiles={true}
        />
      </Link>
    </div>
  </div>
</div>
`;

exports[`CardList testing: is Error shown 1`] = `
<Error
  notice="The request failed!"
/>
`;

exports[`CardList testing: is Loader shown 1`] = `
<Loader
  loader={true}
/>
`;
